src/hist.jl

```diff
@@ -40,46 +40,54 @@ function histrange(v::AbstractArray{T}, n::Integer, closed::Symbol=:left) where
 end
 
 function histrange(lo::F, hi::F, n::Integer, closed::Symbol=:left) where F
-    if hi == lo
-        start = F(hi)
-        step = one(F)
+    closed == :left || closed ==:right || 
+        throw(ArgumentError("closed must be :left or :right"))
+    lo = float(lo)
+    hi = float(hi)
+    if abs(hi-lo) < n*10eps(F)
+        if closed == :left
+            hi = hi + n*10eps(F)
+        elseif closed ==:right
+            lo = lo - n*10eps(F)
+        else
+            # never happens
+        end
+    end
+
+    bw = (F(hi) - F(lo)) / n
+    lbw = log10(bw)
+    if lbw >= 0
+        step = exp10(floor(lbw))
+        r = bw / step
+        if r <= 1.1
+            nothing
```

Comment:
? Why not just omit this condition and start the check with `r <= 2.2`?

Reply:
I did not modify this code path, I will have a look at it

Reply:
Oh sorry, I misread the diff. Don't worry about it for now.